src/overlays/actors/ovl_Obj_Dinner/z_obj_dinner.c

```diff
@@ -26,7 +26,7 @@ extern Gfx D_060011E0[];
 void ObjDinner_Init(Actor* thisx, GlobalContext* globalCtx) {
     ObjDinner* this = THIS;
 
-    if (gSaveContext.perm.isNight != 1 || (CURRENT_DAY == 3 && gSaveContext.perm.weekEventReg[0x16] & 1)) {
+    if (gSaveContext.isNight != true || (CURRENT_DAY == 3 && gSaveContext.weekEventReg[0x16] & 1)) {
```

Comment:
I wonder if this works :thinking: 
```suggestion
    if (!gSaveContext.isNight || (CURRENT_DAY == 3 && gSaveContext.weekEventReg[0x16] & 1)) {
```